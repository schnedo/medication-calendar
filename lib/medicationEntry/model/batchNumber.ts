export default interface BatchNumber {
  readonly number: number;
}

const prefix = "Ch.-B.";
export function format(batchNumber: BatchNumber): string {
  return `${prefix} ${batchNumber.number}`;
}
