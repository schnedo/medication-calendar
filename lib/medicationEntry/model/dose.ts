export default interface Dose {
  readonly amount: number;
}

const unit = "ml";

export function format(dose: Dose): string {
  return `${dose.amount} ${unit}`;
}
