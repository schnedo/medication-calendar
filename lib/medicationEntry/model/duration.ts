export default interface Duration {
  minutes: number;
  hours: number;
}

function formatPart(number: number | undefined) {
  if (number === undefined) {
    return "00";
  }
  const raw = number.toString();
  return raw.length === 1 ? `0${raw}` : raw;
}

export function format(duration: Duration): string {
  return `${formatPart(duration.hours)}:${formatPart(duration.minutes)}`;
}
