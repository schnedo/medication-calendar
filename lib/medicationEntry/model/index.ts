export { default as BatchNumber } from "./batchNumber";
export { default as Dose } from "./dose";
export { default as Duration } from "./Duration";
export { default as Medicament } from "./medicament";
export { default as Medication } from "./medication";
export { default as MedicationEntry } from "./medicationEntry";

export * as Repository from "./repository";
