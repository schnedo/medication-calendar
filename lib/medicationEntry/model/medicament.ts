import BatchNumber from "./batchNumber";

export default interface Medicament {
  readonly name: string;
  readonly batchNumber: BatchNumber;
}
