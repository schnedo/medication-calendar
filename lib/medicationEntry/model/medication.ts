import Dose from "./dose";
import Medicament from "./medicament";
import { v4 } from "uuid";

export default class Medication {
  readonly id: string = v4();

  constructor(readonly dose: Dose, readonly medicament: Medicament) {}
}
