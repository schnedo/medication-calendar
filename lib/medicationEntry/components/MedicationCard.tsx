import { PointerEventHandler, ReactElement } from "react";
import {
  Box,
  ButtonBase,
  Card,
  CardContent,
  makeStyles,
  Typography,
} from "@material-ui/core";
import { formatBatchNumber, formatDose, Medication } from "../model";

const useStyles = makeStyles((theme) => ({
  row: {
    display: "flex",
    justifyContent: "space-between",
  },
  content: {
    flex: "1",
  },
  trailer: {
    marginLeft: theme.spacing(1),
  },
  container: {
    width: "100%",
    textAlign: "justify",
  },
}));

export interface MedicationCardProps {
  medication: Medication;
  onClick?: PointerEventHandler<HTMLButtonElement>;
}

export default function MedicationCard({
  medication: {
    medicament: { name, batchNumber },
    dose,
  },
  onClick,
}: MedicationCardProps): ReactElement {
  const { row, content, trailer, container } = useStyles();
  return (
    <ButtonBase
      onClick={onClick}
      disableRipple={!onClick}
      className={container}
    >
      <Card className={container}>
        <CardContent className={row}>
          <Box className={content}>
            <Typography>{name}</Typography>
            <Typography color="textSecondary">
              {formatBatchNumber(batchNumber)}
            </Typography>
          </Box>
          <Typography className={trailer}>{formatDose(dose)}</Typography>
        </CardContent>
      </Card>
    </ButtonBase>
  );
}
