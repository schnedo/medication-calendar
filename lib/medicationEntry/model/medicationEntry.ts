import Medication from "./medication";
import { BodyMass } from "../../contact/model";
import Duration from "./duration";
import { v4 } from "uuid";

export default class MedicationEntry {
  readonly id: string = v4();

  constructor(
    readonly date: Date,
    readonly medications: Medication[],
    readonly duration: Duration,
    readonly comments: string,
    readonly bodyMass: BodyMass,
  ) {}
}
