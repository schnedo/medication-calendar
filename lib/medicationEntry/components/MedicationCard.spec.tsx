import { render } from "@testing-library/react";
import MedicationCard from "./MedicationCard";
import { Medication } from "../model";
import userEvent from "@testing-library/user-event";

describe("MedicationCard", () => {
  it("should render correctly", async () => {
    expect.hasAssertions();

    const medication: Medication = {
      dose: {
        amount: 200,
      },
      id: "1",
      medicament: {
        name: "medicament name",
        batchNumber: {
          number: 12345,
        },
      },
    };

    const { container } = render(<MedicationCard medication={medication} />);

    expect(container).toMatchSnapshot();
  });

  it("should call onClick when clicked on", async () => {
    expect.hasAssertions();

    const medication: Medication = {
      dose: {
        amount: 200,
      },
      id: "id",
      medicament: {
        name: "name",
        batchNumber: {
          number: 1234,
        },
      },
    };
    const onClick = jest.fn();

    const { getByRole } = render(
      <MedicationCard medication={medication} onClick={onClick} />,
    );

    expect(onClick).not.toHaveBeenCalled();
    userEvent.click(getByRole("button"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
