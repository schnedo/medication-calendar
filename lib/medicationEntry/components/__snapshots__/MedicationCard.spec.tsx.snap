// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationCard should render correctly 1`] = `
<div>
  <button
    class="MuiButtonBase-root makeStyles-container-4"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-container-4 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root makeStyles-row-1"
      >
        <div
          class="MuiBox-root MuiBox-root-5 makeStyles-content-2"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            medicament name
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            Ch.-B. 12345
          </p>
        </div>
        <p
          class="MuiTypography-root makeStyles-trailer-3 MuiTypography-body1"
        >
          200 ml
        </p>
      </div>
    </div>
  </button>
</div>
`;
